### DISCUSSION TITLE
Is there an `onRowClick` event or equivalent?

### DISCUSSION THREAD

**Discussion by lachlancollins** [KevinVandy/material-react-table#47] [Q&A] [answered]
Hi, I'm looking at moving over from [mui-datatables](https://github.com/gregnb/mui-datatables), but the one piece of functionality I couldn't figure out how to migrate was `onRowClick`. This is an event which is called by clicking anywhere in that row. I've tried the row actions and rendering a button in a cell, but neither are perfect for my use case. Thanks!

**Comment by KevinVandy** [comment, marked as answer]
Yes! Read the [Table Event Listener Docs](https://www.material-react-table.com/docs/guides/table-event-listeners) and you will find exactly what you are looking for.

But here is basically what you want

```jsx
<MaterialReactTable
  muiTableBodyRowProps={({ row }) => ({
    onClick: (event) => {
      console.info(event, row.id);
    },
  })}
/>
```

**Reply by lachlancollins** [reply]
Perfect, thank you!
